workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Subscriptions/RequiredSubscriptionsInfoTests.cs

```diff
@@ -0,0 +1,46 @@
+using Improbable.Gdk.Subscriptions;
+using NUnit.Framework;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.EditmodeTests.Subscriptions
+{
+    /// <summary>
+    ///     This tests the correctness of the reflection code that grabs information about Required fields on a class.
+    /// </summary>
+    [TestFixture]
+    public class RequiredSubscriptionsInfoTests
+    {
+        private class RequiredFields
+        {
+            [Require] public World World;
+            [Require] protected EntityId EntityId;
+            [Require] internal ILogDispatcher LogDispatcher;
+            [Require] private Entity entity;
+        }
+
+        [Test]
+        public void RequiredSubscriptionsInfo_finds_all_public_and_private_fields()
+        {
+            var info = new RequiredSubscriptionsInfo(typeof(RequiredFields));
+            Assert.AreEqual(4, info.RequiredTypes.Length);
+
+            Assert.Contains(typeof(World), info.RequiredTypes);
+            Assert.Contains(typeof(EntityId), info.RequiredTypes);
+            Assert.Contains(typeof(ILogDispatcher), info.RequiredTypes);
+            Assert.Contains(typeof(Entity), info.RequiredTypes);
+        }
+
+        private class RequiredStaticFields
+        {
+            [Require] public static World World;
+            [Require] private static Entity entity;
+        }
+
+        [Test]
+        public void RequiredSubscriptionsInfo_ignores_static_fields()
```

Comment:
We might want to make this error. We could also support them, although that seems dangerous.

Reply:
Yeah I'd argue for this to be an error as well. A `static` Require feels incorrect

Reply:
Will open a ticket

Reply:
Raised UTY-1815